core/templates/pages/topics-and-skills-dashboard-page/topics-and-skills-dashboard-page.controller.spec.ts

```diff
@@ -0,0 +1,419 @@
+    });
+
+    it('should reset the filters', function() {
+      const filterObject = (
+        TopicsAndSkillsDashboardFilterObjectFactory.createDefault());
+      expect(ctrl.filterObject).toEqual(filterObject);
+      ctrl.filterObject.sort = 'Newly Created';
+      ctrl.filterObject.keywords = ['keyword1'];
+      ctrl.filterObject.category = 'category1';
+      ctrl.filterObject.status = 'Published';
+      ctrl.resetFilters();
+      expect(ctrl.filterObject).toEqual(filterObject);
+    });
+
+    it('should re-render the filter dropdowns', function() {
+      ctrl.applyFilters();
+      expect(ctrl.select2DropdownIsShown).toBe(false);
+      $timeout.flush();
+      expect(ctrl.select2DropdownIsShown).toBe(true);
+    });
+
+
+    it('should return number from 1 to the range specified', function() {
+      var array = ctrl.generateNumbersTillRange(5);
+      expect(array).toEqual([0, 1, 2, 3, 4]);
+    });
+
+    it('should apply the filters', function() {
+      const topic1 = {
+        is_published: true, name: 'Alpha', category: 'Mathematics',
+        description: 'Alpha description',
+      };
+      const topic2 = {
+        is_published: false, name: 'Alpha2', category: 'Mathematics',
+        description: 'Alp2 desc',
+      };
+      const topic3 = {
+        is_published: false, name: 'Beta', category: 'Mathematics',
+        description: 'Beta description',
+      };
+      const topic4 = {
+        is_published: true, name: 'Gamma', category: 'Mathematics',
+        description: 'Gamma description',
+      };
+      ctrl.filterObject = (
+        TopicsAndSkillsDashboardFilterObjectFactory.createDefault());
+      ctrl.totalTopicSummaries = [topic1, topic2, topic3, topic4];
+
+      ctrl.applyFilters();
+      expect(ctrl.topicSummaries).toEqual(ctrl.totalTopicSummaries);
+      expect(ctrl.displayedTopicSummaries).toEqual(ctrl.totalTopicSummaries);
+      expect(ctrl.currentCount).toEqual(4);
+      expect(ctrl.pageNumber).toEqual(0);
+
+      ctrl.filterObject.status = 'Published';
+      ctrl.applyFilters();
+      expect(ctrl.topicSummaries).toEqual([topic1, topic4]);
+      expect(ctrl.displayedTopicSummaries).toEqual([topic1, topic4]);
+      expect(ctrl.currentCount).toEqual(2);
+      expect(ctrl.pageNumber).toEqual(0);
+
+      ctrl.filterObject.keywords = ['gamm'];
+      ctrl.applyFilters();
+      expect(ctrl.topicSummaries).toEqual([topic4]);
+      expect(ctrl.displayedTopicSummaries).toEqual([topic4]);
+      expect(ctrl.currentCount).toEqual(1);
+      expect(ctrl.pageNumber).toEqual(0);
+    });
+
+    it('should call goToPageNumber when refreshPagination is called',
+      function() {
+        var changePageSpy = spyOn(ctrl, 'goToPageNumber');
+        ctrl.refreshPagination();
+        expect(changePageSpy).toHaveBeenCalledWith(0);
+      });
+
+    it('should call initDashboard on reinitialized event', function() {
+      var initDashboardSpy = spyOn(ctrl, '_initDashboard');
```

Comment:
Just wrapping up, I think it's fine to spy on private methods. If you were calling this private method here, then it would be a problem.

Reply:
Yeah, I was also thinking the same thing. Thanks!

@DubeySandeep @seanlip I think this resolves your query too.